src/components/carousel/index.js

```diff
@@ -0,0 +1,209 @@
+
+import classnames from "classnames"
+import withWindow from "../../utils/withWindow"
+import useCurrentWitdh from "../../utils/useCurrentWitdh"
+
+//
+// images: array of image. { url, title }
+//
+function Carousel({ images = [], height, onClickRoot }) {
+  const [width, setWidth] = React.useState(0)
+  const [isHoverMode, setIsHoverMode] = React.useState(false)
+  const [currentIndex, setCurrentIndex] = React.useState(0)
+
+  const windowWidth = useCurrentWitdh()
+
+  const nbImages = images.length
+
+  const rootRef = React.useRef(null)
+
+  //
+  // set isHoverMode to true and hide it after some time
+  //
+  let timeoutId = null
+  const activateHoverMode = () => {
+    setIsHoverMode(true)
+    clearTimeout(timeoutId)
+    timeoutId = setTimeout(() => {
+      setIsHoverMode(false)
+    }, 600)
+  }
+
+  //
+  // currentIndex utilities
+  //
+
+  const previous = () => {
+    if (currentIndex > 0) {
+      setCurrentIndex(currentIndex - 1)
+    } else {
+      setCurrentIndex(nbImages - 1)
+    }
+  }
+
+  const next = () => {
+    if (currentIndex + 1 === nbImages) {
+      setCurrentIndex(0)
+    } else {
+      setCurrentIndex(currentIndex + 1)
+    }
+  }
+
+  //
+  // window resize handling
+  //
+
+  // retrieve our carousel width
+  // we use the windowWidth to recalculate our element width when the window width changed
+  // (is resized).
+  React.useEffect(() => {
+    setWidth(rootRef.current?.offsetWidth || 0)
+  }, [rootRef.current, setWidth, windowWidth])
+
+  //
+  // keyboard shortcuts handling
+  //
+
+  const handleKeyDown = React.useCallback(
+    e => {
+      // arrow up/down button should select next/previous list element.
+      // we also use vim-keys (h,j,k,l)
+      if (
+        // up
+        e.keyCode === 38 ||
+        // left
+        e.keyCode === 37 ||
+        // h
+        e.keyCode === 72 ||
+        // k
+        e.keyCode === 75
+      ) {
+        previous()
+      } else if (
+        // down
+        e.keyCode === 40 ||
+        // right
+        e.keyCode === 39 ||
+        // j
+        e.keyCode === 74 ||
+        // l
+        e.keyCode === 76
+      ) {
+        next()
+      }
+
+      activateHoverMode()
+    },
+    [setIsHoverMode, setCurrentIndex, currentIndex]
+  )
+
+  // TODO: for now we add a listener for the whole document, maybe we would want to reduce it only
+  //  when we hover the carousel?
+  React.useEffect(() => {
+    document.addEventListener("keydown", handleKeyDown)
+
+    return () => {
+      document.removeEventListener("keydown", handleKeyDown)
+    }
+  }, [handleKeyDown])
+
+  //
+  //
+  //
+
+  return (
+    <div
+      ref={rootRef}
+      role="presentation"
+      className="relative cursor-pointer text-center overflow-hidden"
+      // allow parent to define a custom onClickRoot. By default we display the next image on click
+      onClick={onClickRoot || next}
+      onMouseEnter={activateHoverMode}
+      onMouseLeave={() => setIsHoverMode(false)}
```

Comment:
Using the carousel, I expected hover mode to be activated when moving the mouse within the carousel (after it has faded away). Is this easy to add here?

Reply:
I am sorry but I am not sure to understand what you mean here.

My guess is that you want the picture to change not when we put the mouse on top of the image but when we already are on top of it and move the mouse around ?

Reply:
Sorry, I was saying that i would expect the little black bar at the bottom to show any time I move the mouse within the carousel. Right now it shows when I enter, then disappears until I re-enter the carousel.

Reply:
I added this behavior because the bar is on top of the image. Otherwise  when we hover an image we will always have the bar to be visible. 

I though it would be better to hide it after some time without moving the mouse. The current behavior is not exactly when moving the mouse but when moving to a new image.

Maybe we could have an in-between when:
- the bar is hidden after x time without moving the mouse
- the bar is shown again when me move the mouse 
But it seem complicated for almost nothing.

Reply:
👍 I prefer the current behavior. 

If we feel like it is confusing we can re-evaluate in a couple of weeks.